### PR TITLE
Update deps (latest) on released prerelease packages in the client release group

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Update deps (latest) on released prerelease packages in the client release group for minor release of 2.0.0-internal.3.1.0

Other changes which were causing build failures:
1.) File failed policy check: server/routerlicious/pnpm-lock.yaml Invalid pnpm enforcement preinstall script "node ./scripts/only-pnpm.cjs" in package.json - Fix was to remove "./" from preinstall command:  "node ./scripts/only-pnpm.cjs" in package.json. This is same as in client repo.

2.) Needed to delete: server/routerlicious/package-lock.json because we now have pnpm-lock.yaml.

These seems to have been caused by update in build tools which is happening in this PR from "^0.9.0-127618" to "^0.9.0"

### CODE REVIEW

File: server/routerlicious/package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "docker-build": "docker-compose build",
     "docker-clean": "docker-compose down --rmi local -v",
     "format": "lerna run format --parallel --stream",
-    "preinstall": "node ./scripts/only-pnpm.cjs",
```
Comment (CraigMacomber) [inline server/routerlicious/package.json:33]: why this change? Starting paths with ./ when they are relative to the working directory is good practice as it avoids several possible issues (like accidently starting the path at the root or somewhere in the search path).

Did this cause an issue? If so, note this change in the PR description since it can easilly hid with all the other file changes in this PR

**Comment by CraigMacomber** [comment]
Deleting server/routerlicious/package-lock.json seems like an unrelated change. It probably shouldn't be in this PR, but if it is, ity should be in the description along with a note of why its included.

File: server/routerlicious/package.json (hunk 1 above)
Comment (jatgarg) [reply server/routerlicious/package.json:33]: Yes it caused this issue: file failed policy check: server/routerlicious/pnpm-lock.yamlInvalid pnpm enforcement preinstall script "node ./scripts/only-pnpm.cjs" in package.json
I checked our client repository and saw that it does not had this issue because there was no "./" in there.

**Comment by jatgarg** [comment]
> Deleting server/routerlicious/package-lock.json seems like an unrelated change. It probably shouldn't be in this PR, but if it is, ity should be in the description along with a note of why its included.

Adding in the PR description.

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -10311,7 +10311,7 @@ packages:
     resolution: {integrity: sha512-RAmZReFuKCKx0Rs5C0nVJwKomAHUHn+gGP/YvbEsXQWu0sXoncEUZa71UqlfCPVXa/0MkYOIbCXSQdOcuRrHgw==}
     peerDependencies:
       graphql: ^14.0.0 || ^15.0.0
-      react: ^16.8.0 || ^17.0.0
+      react: ^16.8.0 || ^17.0.0 || 17.0.2
```
Comment (CraigMacomber) [inline pnpm-lock.yaml:10314]: this change is interesting. I wonder why it happened.

Comment (jatgarg) [reply pnpm-lock.yaml:10314]: Because we have a peer dep in package.json:
	"pnpm": {
		"overrides": {
			"@types/node": "^14.18.36"
		},
		"peerDependencyRules": {
			"allowedVersions": {
				"react": "17.0.2",
				"react-dom": "17.0.2",
				"webpack": "5.75"
			},
			"ignoreMissing": [
				"fluid-framework"
			]
		}
	}

File: server/routerlicious/package.json (hunk 1 above)
Comment (jatgarg) [reply server/routerlicious/package.json:33]: Added

File: pnpm-lock.yaml (hunk 2 above)
Comment (CraigMacomber) [reply pnpm-lock.yaml:10314]: I have tried and failed to figure that out. It repros for me and is probably fine

**Comment by CraigMacomber** [comment]
Were the extra changes needed because this change updates the version of build tools that is used, and it objected? If so, it would be good to include that as part of why you did those changes, because currently its unclear why they are part of this PR instead of a different one.

**Review by CraigMacomber — APPROVED** [review]
